src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -408,6 +408,16 @@ public static bool IsAsyncSupportingFunctionSyntax([NotNullWhen(returnValue: tru
                 || node.IsKind(SyntaxKind.LocalFunctionStatement);
         }
 
+        public static TypeSyntax? GetFunctionReturnTypeSyntax(this SyntaxNode node)
+        {
+            return node switch
+            {
+                MethodDeclarationSyntax methodDeclaration => methodDeclaration.ReturnType,
+                LocalFunctionStatementSyntax localFunctionStatement => localFunctionStatement.ReturnType,
```

Comment:
💡 I didn't see tests specifically for local functions; would be good to add them.

Reply:
@sharwell I recall I added them as integration tests